tests/api/cluster/test_app.py

```diff
@@ -0,0 +1,74 @@
+from unittest.mock import MagicMock
+
+import pytest
+
+from controlpanel.api.cluster import (
+    BASE_ROLE_POLICY,
+    App,
+)
+from controlpanel.api import models
+
+from django.conf import settings
+
+
+@pytest.fixture
+def app_model():
+    return models.App(
+        slug = "slug",
+        repo_url = "https://gitpub.example.com/test-repo",
+    )
+
+@pytest.fixture
+def app(app_model):
+    return App(app_model)
+
+
+def test_app_iam_role_name(app, app_model):
+    expected_iam_role_name = f"{settings.ENV}_app_{app_model.slug}"
+    assert app.iam_role_name == expected_iam_role_name
+
+
+def test_app_create_iam_role(aws, app):
+    app.create_iam_role()
+
+    aws.create_role.assert_called_with(
+        app.iam_role_name,
+        BASE_ROLE_POLICY,
+    )
+
+
+def test_app_delete_iam_role(aws, app):
+    app.delete_iam_role()
+    aws.delete_role.assert_called_with(app.iam_role_name)
+
+
+def test_app_url_when_no_ingresses(k8s_client, app, app_model):
+    k8s_client.ExtensionsV1beta1Api.list_namespaced_ingress.return_value.items = []
+
+    assert app.url == ""
+    k8s_client \
+        .ExtensionsV1beta1Api \
+        .list_namespaced_ingress \
+        .assert_called_once_with("apps-prod", label_selector="repo=test-repo")
```

Comment:
I'm pretty sure you can and should remove these `\`s - but use [Black](https://black.readthedocs.io/en/stable/index.html) as the source of truth on formatting

Reply:
You can break the line after `(` like:
```python
    k8s_client.ExtensionsV1beta1Api.list_namespaced_ingress.assert_called_once_with(
        "apps-prod",
        label_selector="repo=test-repo",
    )
```
The long line is <88 chars, which is the default max line-length in Black.

Reply:
@r4vi (who was there when I was writing this) told me to tell you I run this through black :trollface: 

I'll play with it. (no wonder Go's autoformatting is everyone's favourite feature 😎)